src/Voron/StorageEnvironment.cs

```diff
@@ -1687,16 +1687,24 @@ internal ApplyLogsToDataFileState TryGetLatestEnvironmentStateToFlush(long uptoT
             while (true)
             {
                 if (_transactionsToFlush.TryPeek(out var maybe) == false ||
-                    maybe.TransactionId >= uptoTxIdExclusive)
+                    maybe.EnvStateRecord.TransactionId >= uptoTxIdExclusive)
```

Comment:
Can this be null?

Reply:
No, we set it to `null` _after_ dequeuing. So as long as it's in the queue it must not be `null`.